From patchwork Thu May 21 05:57:49 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: =?utf-8?b?5b2t5rWpKFJpY2hhcmQp?=
 <richard.peng@oppo.com>
X-Patchwork-Id: 11562279
Return-Path: <SRS0=9zq6=7D=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 956BD912
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 21 May 2020 05:57:59 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 7789720825
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 21 May 2020 05:57:59 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=oppoglobal.onmicrosoft.com
 header.i=@oppoglobal.onmicrosoft.com header.b="Rbp++ErR"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1728083AbgEUF54 (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 21 May 2020 01:57:56 -0400
Received: from mail-eopbgr1310070.outbound.protection.outlook.com
 ([40.107.131.70]:11440
        "EHLO APC01-SG2-obe.outbound.protection.outlook.com"
        rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org with ESMTP
        id S1726869AbgEUF5z (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 21 May 2020 01:57:55 -0400
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=KyRyeaQH+NSrNqZFjECHNd69s7ltflQW/H3rjIbjZFwq9tJ7EgSCoKQhnnbbIbEWMiixdPJyQas1oYkc455nvp0UqlgJCQ4CO41R9I4DdYH7VrO0FlYXW5MerWEa8uzSG2xEECg6YUf1/THcwmq+b26Du0yUCHLcLVg9v2g/rZnD2hoi2tI90T1J/Vh5do4UOJfUOyDJPyhQPJVaERvX1QPjq/yOEzHfEeemBHMdVpzfHPU7QJWf7WNw+BRY2QHTpepSUUNiSmVl4Y+4rCcrwqpaQf857hRizep2smHU6PvXE7xFYQRsT7Iw+6qHnpxdJCc7p+jiub6lwl/XFfjExA==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=9zKc6D+OtGRV1jtx43O1KVZP6EL0f/3uvVP4n8uZC50=;
 b=Gy/oIsYCVf4Ppmp5aT0R34fCmCsMoGzUs+tl+oXcmmR8RmnDpFMLLa6D2pheVZ3hqFcmldiswADaWexfM5ocR5KZdxmF0XhxaWW5VKyOiXkuMTba926AkfFLmxxIAGKOiE5uiXc9dv9osnr3zTxFQsGCfe1NDErPdSYt2h+irTe4DVTP+GBZ2qr06a7fSSeQS59ZNnmOE9rQmM1aUkNymPEWkjTaBRgXZmkLIV8GAMbacqhHZi8tKpXmnAzNDvKwEtyv3NzCns4dznBNHT9VP2jYy161yOvZeoBkYB9HZ2mCj7m1d7CxH8WaB3w36d1MXqiJA1DcORO1aFvZSwQVkA==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oppo.com; dmarc=pass action=none header.from=oppo.com;
 dkim=pass header.d=oppo.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oppoglobal.onmicrosoft.com; s=selector1-oppoglobal-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=9zKc6D+OtGRV1jtx43O1KVZP6EL0f/3uvVP4n8uZC50=;
 b=Rbp++ErRhn/sPpvVYXvUK3KiJHlDZR4qS90V7G5kpKvglr0vVvA7d6f3UNJbZi1y8w0fDYtzjb81g+RjBlL88Q4x2yDgYgYZtA3odlrCZtUo8e14+4EEGALggIPqaYLTJcm/k7aLpExu3PYrFxk8a0hoyIo0P5PR1pRk1mNvb0s=
Received: from HKAPR02MB4291.apcprd02.prod.outlook.com (2603:1096:203:d3::12)
 by HKAPR02MB4403.apcprd02.prod.outlook.com (2603:1096:203:d5::8) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3021.23; Thu, 21 May
 2020 05:57:49 +0000
Received: from HKAPR02MB4291.apcprd02.prod.outlook.com
 ([fe80::80ca:c698:988e:a999]) by HKAPR02MB4291.apcprd02.prod.outlook.com
 ([fe80::80ca:c698:988e:a999%5]) with mapi id 15.20.3021.020; Thu, 21 May 2020
 05:57:49 +0000
From: =?eucgb2312_cn?b?xe26xihSaWNoYXJkKQ==?= <richard.peng@oppo.com>
To: Paolo Bonzini <pbonzini@redhat.com>
CC: "linux-kernel@vger.kernel.org" <linux-kernel@vger.kernel.org>,
        "kvm@vger.kernel.org" <kvm@vger.kernel.org>
Subject: [PATCH v3] kvm/x86 : Remove redundant function implement
Thread-Topic: [PATCH v3] kvm/x86 : Remove redundant function implement
Thread-Index: AdYvNGncuTv2pa/PRZeWvNnr93KTAw==
Date: Thu, 21 May 2020 05:57:49 +0000
Message-ID: 
 <HKAPR02MB4291D5926EA10B8BFE9EA0D3E0B70@HKAPR02MB4291.apcprd02.prod.outlook.com>
Accept-Language: zh-CN, en-US
Content-Language: zh-CN
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
authentication-results: redhat.com; dkim=none (message not signed)
 header.d=none;redhat.com; dmarc=none action=none header.from=oppo.com;
x-originating-ip: [58.255.79.102]
x-ms-publictraffictype: Email
x-ms-office365-filtering-correlation-id: 5b49bb71-8b06-4612-6052-08d7fd4be4ac
x-ms-traffictypediagnostic: HKAPR02MB4403:
x-microsoft-antispam-prvs: 
 <HKAPR02MB440328797D61BB3BD7D2B0CCE0B70@HKAPR02MB4403.apcprd02.prod.outlook.com>
x-ms-oob-tlc-oobclassifiers: OLM:119;
x-forefront-prvs: 041032FF37
x-ms-exchange-senderadcheck: 1
x-microsoft-antispam: BCL:0;
x-microsoft-antispam-message-info: 
 IQyWPh9tKCanaO5483V3q4Eyoijqxyy44KSPS7LiaIddnjBR7oAs39IQPP561tQR7i7PK8pQVP6Sp0W/Had1LeS6IIjIKfaMv/3ohfT2/rJCR9RJ5OFiE/0xsPIlkokRS2sGxMq4FHYJ857HBi4N6pBvc/1GtTZTUP9mU2BSlDDD6vVvf5p62OvAs1bMvLQga5s9qa38npZirqsIIjv4GVOJsjwkJzfWPyd6o0AHjfOY1INkC2q8iri52sRbbIlTB33ZzKnkrnbTB7EvgRKCt3N4oLhQMPARe2YJJ+4H+rxbDfh933lJwCffq4m1c3xIDrGgBqjll+ie9udq6eT2VxKl2QwScmvhP1T/C3W7eOwSz60NPEvIEUihrE49fzGX
x-forefront-antispam-report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:HKAPR02MB4291.apcprd02.prod.outlook.com;PTR:;CAT:NONE;SFTY:;SFS:(396003)(346002)(136003)(39860400002)(376002)(366004)(4326008)(85182001)(86362001)(8676002)(5660300002)(8936002)(66446008)(64756008)(9686003)(55016002)(6916009)(71200400001)(66476007)(186003)(6506007)(478600001)(33656002)(52536014)(66946007)(76116006)(54906003)(66556008)(316002)(26005)(7696005)(2906002)(11606004);DIR:OUT;SFP:1101;
x-ms-exchange-antispam-messagedata: 
 pFto/N3jMm9e9mxV+w54/p7PEit6mnK6d6HswKMdneQzr6QohH3h23fxOMn/3puelH1qgdLy42KR/jY2fGG+7AahSwia9i4J63ef68HECMrR4NS4ck5GGhelOOzMmhI1y0Z3Iw3svFW6YonXuBByVbwyXrTPfPNC/xgakLEvcUf5QnScGmUlrjpzbgOr88hKHI2lbHh1IMu7z6IhVSV+09a4aCC4z936iQS1vDgoNMDctQzQ0dVn8WPcnLzGoOb99yjtzK8IhWATeBW+OH1QVVyxbAv+0qCbIx9PjWgaJ+piRna4ZJGnOEd/JK3Sb6KpmjMPjQsmVobHZ6tKCZXC7B1M2YmnFXVhotem1/mDJE1PpR9dNyUWoTlR5dngCgyJ+DPptPolZed+77fsOvHoY5TfpThY6lnBtJNMrOs4KmXyjDb/7VE+JqxcCoMWpy85drnHp4if8W4cdEjJy4H6+CYRO2JzXVZ/Pf/pS9Bq1EE=
x-ms-exchange-transport-forked: True
MIME-Version: 1.0
X-OriginatorOrg: oppo.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 5b49bb71-8b06-4612-6052-08d7fd4be4ac
X-MS-Exchange-CrossTenant-originalarrivaltime: 21 May 2020 05:57:49.4960
 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: f1905eb1-c353-41c5-9516-62b4a54b5ee6
X-MS-Exchange-CrossTenant-mailboxtype: HOSTED
X-MS-Exchange-CrossTenant-userprincipalname: 
 jeWj00RCj804FikMJx1Fwq9fnBGy9qElZiiwHgJF19dIl2IWEezXd1xF/sKVYHeANCsNIc/dK5hPgAQLwy1Zng==
X-MS-Exchange-Transport-CrossTenantHeadersStamped: HKAPR02MB4403
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

pic_in_kernel(),ioapic_in_kernel() and irqchip_kernel() have the
same implementation.

Signed-off-by: Peng Hao <richard.peng@oppo.com>
---
 arch/x86/kvm/ioapic.h  |  8 ++------
 arch/x86/kvm/irq.h     | 14 ++++----------
 arch/x86/kvm/lapic.c   |  1 +
 arch/x86/kvm/mmu/mmu.c |  1 +
 arch/x86/kvm/x86.c     |  1 +
 5 files changed, 9 insertions(+), 16 deletions(-)

--
2.7.4

diff --git a/arch/x86/kvm/ioapic.h b/arch/x86/kvm/ioapic.h
index 2fb2e3c..7a3c53b 100644
--- a/arch/x86/kvm/ioapic.h
+++ b/arch/x86/kvm/ioapic.h
@@ -5,7 +5,7 @@
 #include <linux/kvm_host.h>

 #include <kvm/iodev.h>
-
+#include "irq.h"
 struct kvm;
 struct kvm_vcpu;

@@ -108,11 +108,7 @@ do {\

 static inline int ioapic_in_kernel(struct kvm *kvm)
 {
-int mode = kvm->arch.irqchip_mode;
-
-/* Matches smp_wmb() when setting irqchip_mode */
-smp_rmb();
-return mode == KVM_IRQCHIP_KERNEL;
+return irqchip_kernel(kvm);
 }

 void kvm_rtc_eoi_tracking_restore_one(struct kvm_vcpu *vcpu);
diff --git a/arch/x86/kvm/irq.h b/arch/x86/kvm/irq.h
index f173ab6..e133c1a 100644
--- a/arch/x86/kvm/irq.h
+++ b/arch/x86/kvm/irq.h
@@ -16,7 +16,6 @@
 #include <linux/spinlock.h>

 #include <kvm/iodev.h>
-#include "ioapic.h"
 #include "lapic.h"

 #define PIC_NUM_PINS 16
@@ -66,15 +65,6 @@ void kvm_pic_destroy(struct kvm *kvm);
 int kvm_pic_read_irq(struct kvm *kvm);
 void kvm_pic_update_irq(struct kvm_pic *s);

-static inline int pic_in_kernel(struct kvm *kvm)
-{
-int mode = kvm->arch.irqchip_mode;
-
-/* Matches smp_wmb() when setting irqchip_mode */
-smp_rmb();
-return mode == KVM_IRQCHIP_KERNEL;
-}
-
 static inline int irqchip_split(struct kvm *kvm)
 {
 int mode = kvm->arch.irqchip_mode;
@@ -93,6 +83,10 @@ static inline int irqchip_kernel(struct kvm *kvm)
 return mode == KVM_IRQCHIP_KERNEL;
 }

+static inline int pic_in_kernel(struct kvm *kvm)
+{
+return irqchip_kernel(kvm);
+}
 static inline int irqchip_in_kernel(struct kvm *kvm)
 {
 int mode = kvm->arch.irqchip_mode;
diff --git a/arch/x86/kvm/lapic.c b/arch/x86/kvm/lapic.c
index 9af25c9..de4d046 100644
--- a/arch/x86/kvm/lapic.c
+++ b/arch/x86/kvm/lapic.c
@@ -36,6 +36,7 @@
 #include <linux/jump_label.h>
 #include "kvm_cache_regs.h"
 #include "irq.h"
+#include "ioapic.h"
 #include "trace.h"
 #include "x86.h"
 #include "cpuid.h"
diff --git a/arch/x86/kvm/mmu/mmu.c b/arch/x86/kvm/mmu/mmu.c
index 8071952..6133f69 100644
--- a/arch/x86/kvm/mmu/mmu.c
+++ b/arch/x86/kvm/mmu/mmu.c
@@ -16,6 +16,7 @@
  */

 #include "irq.h"
+#include "ioapic.h"
 #include "mmu.h"
 #include "x86.h"
 #include "kvm_cache_regs.h"
diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index d786c7d..c8b62ac 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -18,6 +18,7 @@

 #include <linux/kvm_host.h>
 #include "irq.h"
+#include "ioapic.h"
 #include "mmu.h"
 #include "i8254.h"
 #include "tss.h"
